pkg/apis/cluster/v1alpha1/common_types.go

```diff
@@ -39,5 +42,25 @@ type ProviderConfig struct {
 // ProviderConfigSource represents a source for the provider-specific
 // resource configuration.
 type ProviderConfigSource struct {
-	// TODO(roberthbailey): Fill these in later
+	// The machine class from which the provider config should be sourced.
+	// +optional
+	MachineClass *MachineClassRef `json:machineClass,omitempty`
+}
+
+// MachineClassRef is a reference to the MachineClass object. Controllers should find the right MachineClass using this reference.
+type MachineClassRef struct {
```

Comment:
mvladev: What about namespace?

roberthbailey: Storage classes aren't namespaced, and I used that same pattern here. But it's worth discussing whether they should be part of a namespace.

Reply:
I have entertained the namespaces- should be helpful especially keeping in mind the sensitivity of the data in machine-classes and our approach of isolating the clusters based on namespaces.

Reply:
StorageClass are basically templates.  Here, we're using namespace for isolating actual clusters.  Do we need to also isolate machine templates?  I don't see a need for namespace here.

Reply:
As mentioned, machine-classes going forward would be expected to have specific details towards machines and in-turn clusters, and also sensitive information of underlying infra. Separating machine-classes via namespaces will help to avoid unintended consumers of it.

Reply:
tend to agree with @hardikdr on keeping machine class as namespaced object.